src/invariant/EventsAreConsistentWithEntryDiffs.cpp

```diff
@@ -954,7 +138,7 @@ calculateDeltaBalance(AggregatedEvents const& agg, LedgerEntry const* current,
         auto entryDiff = (current ? current->data.account().balance : 0) -
                          (previous ? previous->data.account().balance : 0);
 
-        return entryDiff == eventDiff ? ""
+        return entryDiff == (int64)eventDiff.lo ? ""
```

Comment:
Why is `entryDiff` being compared to `lo` and not the full `eventDiff`?

Reply:
the eventDiff is an i128 and we are comparing it with an int64 (from classic), so it's "downcasting". I can separate this into a helper method to make it more clear.

Reply:
It is safer upcast the int64 ledger entry diff into an I128 (as you originally had) instead of downcasting. 
I've fixed it by adding a new bridge function for conversion and adapted the comparison logic here. 
This also lets us get rid of the ad-hoc conversion from i64 to ScVal method in eventManager.